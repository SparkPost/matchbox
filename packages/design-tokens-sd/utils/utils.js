function toCamel(string) {
  return string
    .replace('-', ' ')
    .replace(/(?:^\w|[A-Z]|\b\w)/g, (word, index) => {
      return index === 0 ? word.toLowerCase() : word.toUpperCase();
    })
    .replace(/\s+/g, '');
}

function toSnake(string) {
  return string
    .replace(/\W+/g, ' ')
    .split(/ |\B(?=[A-Z])/)
    .map((word) => word.toLowerCase())
    .join('_');
}

module.exports = {
  toCamel,
  toSnake,
};
