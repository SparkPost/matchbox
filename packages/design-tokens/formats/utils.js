function getName(name, prefix) {
  const noPrefix = name.replace(`${prefix}-`, '').split('-');
  return noPrefix.pop();
}

function getPalette(name, prefix) {
  return name.replace(`${prefix}-`, '').split('-').shift();
}

function groupByPalette(props, prefix) {
  const withPalette = props.map((prop) => ({
    ...prop,
    palette: getPalette(prop.name, prefix)
  }
  ));

  const groupedByPalette = withPalette.reduce((acc, prop) => {
    if (!acc[prop.palette]) {
      acc[prop.palette] = [prop];
    } else {
      acc[prop.palette].push(prop);
    }

    return acc;
  }, {});

  return groupedByPalette;
}

function kebabToCamel(str) {
  return str.replace(/-([a-z|0-9])/g, (g) => g[1].toUpperCase());
}

function kebabToFriendly(str) {
  const r = str.replace(/-([a-z|0-9])/g, (g) => ` ${g[1].toUpperCase()}`);
  return r.charAt(0).toUpperCase() + r.slice(1);
}
/**
 * Returns a camel cased name from a kebab cased string,
 * with a period(.) before token variant
 * @example
 *   getCommonJsName('font-size-500', 'font-size')
 *   // 'fontsize_500'
 */
function getCommonJsName(str, type) {
  const variantOrPalette = str.replace(`${type}-`, '');
  const parts = variantOrPalette.split('-');

  const prefix = kebabToCamel(type);
  const getVariant = (str) => str === 'base' ? '' : `_${str}`;

  if (parts.length === 1) {
    return `${prefix}${getVariant(parts.pop())}`;
  }

  return `${prefix}_${parts[0]}${getVariant(parts.pop())}`;
}

module.exports = {
  getPalette,
  getName,
  getCommonJsName,
  groupByPalette,
  kebabToCamel,
  kebabToFriendly
};
