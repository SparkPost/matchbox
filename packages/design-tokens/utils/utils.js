function toCamel(string) {
  return string
    .replace('-', ' ')
    .replace(/(?:^\w|[A-Z]|\b\w)/g, (word, index) =>
      index === 0 ? word.toLowerCase() : word.toUpperCase(),
    )
    .replace(/\s+/g, '');
}

function toSnake(string) {
  return string
    .replace(/\W+/g, ' ')
    .split(/ |\B(?=[A-Z])/)
    .map((word) => word.toLowerCase())
    .join('_');
}

function toFriendly(string) {
  return string
    .toLowerCase()
    .replace('-', ' ')
    .split(' ')
    .map((s) => s.charAt(0).toUpperCase() + s.substring(1))
    .join(' ');
}

module.exports = {
  toCamel,
  toSnake,
  toFriendly,
};
