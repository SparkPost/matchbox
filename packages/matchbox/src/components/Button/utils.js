import React from 'react';
import Button from './Button';
import { filterByVisible } from '../../helpers/array';

export function buttonsFrom(actions, overrides) {
  const filteredActions = filterByVisible(actions);

  if (filteredActions.length) {
    return <Button.Group>{filteredActions.map((action, key) => buttonFrom(action, overrides, key))}</Button.Group>;
  }
}

export function buttonFrom({ content, ...action }, overrides, key) {
  return (
    <Button
      key={key}
      children={content}
      {...action}
      {...overrides}
    />
  );
}
