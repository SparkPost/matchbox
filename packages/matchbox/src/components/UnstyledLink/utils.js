import React from 'react';
import UnstyledLink from './UnstyledLink';

export function linkFrom({ content, ...action }, key) {
  return (
    <UnstyledLink
      key={key}
      children={content}
      {...action}
    />
  );
}
