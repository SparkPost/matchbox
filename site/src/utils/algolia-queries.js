function pageToAlgoliaRecord({ node: { id, frontmatter, fields, ...rest } }) {
  return {
    objectID: id,
    ...frontmatter,
    ...fields,
    ...rest
  };
}

const pageQuery = `{
  allMdx {
    edges {
      node {
        id
        fields {
          slug
        }
        frontmatter {
          title
          category
        }
        excerpt(pruneLength: 200)
      }
    }
  }
}`;

const queries = [
  {
    query: pageQuery,
    transformer: ({ data }) => {
      return data.allMdx.edges.map(pageToAlgoliaRecord);
    }, // optional
    indexName: 'matchbox', // overrides main index name, optional
    settings: { attributesToSnippet: [`excerpt:20`] }
    // matchFields: ['slug', 'modified'] // Array<String> overrides main match fields, optional
  }
];

module.exports = queries;
