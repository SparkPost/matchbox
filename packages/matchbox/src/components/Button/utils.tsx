import React from 'react';
import Button from './Button';
import { filterByVisible } from '../../helpers/array';
import type { BaseProps } from './Button';

interface ButtonPropsWithContent extends BaseProps {
  content?: React.ReactNode;
}

export function buttonsFrom(actions: ButtonPropsWithContent[], overrides: BaseProps) {
  const filteredActions = filterByVisible(actions);

  if (filteredActions.length) {
    return (
      <Button.Group>
        {filteredActions.map((action: ButtonPropsWithContent, key: React.Key) =>
          buttonFrom(action, overrides, key),
        )}
      </Button.Group>
    );
  }
}

export function buttonFrom(
  { content, ...action }: ButtonPropsWithContent,
  overrides: BaseProps,
  key: React.Key,
) {
  return <Button key={key} children={content} {...action} {...overrides} />; // eslint-disable-line
}
