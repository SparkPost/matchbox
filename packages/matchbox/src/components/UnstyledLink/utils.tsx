import React from 'react';
import UnstyledLink from './UnstyledLink';
import type { UnstyledLinkBaseProps } from './UnstyledLink';

interface LinkFromTypes extends UnstyledLinkBaseProps {
  content?: React.ReactNode;
  children?: React.ReactNode;
}

export function linkFrom(
  { content, children, ...action }: LinkFromTypes,
  key?: React.Key,
): JSX.Element {
  return (
    <UnstyledLink key={key} {...action}>
      {content}
      {children}
    </UnstyledLink>
  );
}
