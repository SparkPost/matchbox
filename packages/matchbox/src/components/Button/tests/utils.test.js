import React from 'react';
import { buttonFrom, buttonsFrom } from '../utils';

describe('Button Utils', () => {

  describe('buttonFrom', () => {
    it('renders correctly', () => {
      expect(buttonFrom({ content: <span>button content</span>, disabled: false })).toMatchSnapshot();
    });

    it('passes any custom overrides', () => {
      expect(buttonFrom({ content: <span>button content</span> }, { foo: 'bar', type: 'submit' })).toMatchSnapshot();
    });
  });

  describe('buttonsFrom', () => {
    it('renders correctly', () => {
      const actions = [
        { content: <span>button 1 content</span>, action: { content: <span>foo</span> }},
        { content: <span>button 2 content</span>, action: {}}
      ];

      expect(buttonsFrom(actions)).toMatchSnapshot();
    });

    it('returns nothing without actions', () => {
      const actions = [];
      expect(buttonsFrom(actions)).toBe(undefined);
    });
  });
});
