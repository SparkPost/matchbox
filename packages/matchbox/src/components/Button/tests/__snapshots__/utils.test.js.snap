// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Utils buttonFrom passes any custom overrides 1`] = `
<Button
  foo="bar"
  size="default"
  type="submit"
>
  <span>
    button content
  </span>
</Button>
`;

exports[`Button Utils buttonFrom renders correctly 1`] = `
<Button
  disabled={false}
  size="default"
>
  <span>
    button content
  </span>
</Button>
`;

exports[`Button Utils buttonsFrom renders correctly 1`] = `
<Button.Group>
  <Button
    action={
      Object {
        "content": <span>
          foo
        </span>,
      }
    }
    size="default"
  >
    <span>
      button 1 content
    </span>
  </Button>
  <Button
    action={Object {}}
    size="default"
  >
    <span>
      button 2 content
    </span>
  </Button>
</Button.Group>
`;
